src/configure/mod.rs

```diff
@@ -593,40 +613,53 @@ async fn try_imap_one_param(context: &Context, param: &LoginParam, imap: &mut Im
     }
 
     if context.shall_stop_ongoing().await {
-        bail!("Interrupted");
+        return Ok(());
     }
 
     bail!("Could not connect: {}", inf);
 }
 
 async fn try_smtp_connections(
     context: &Context,
-    mut param: &mut LoginParam,
+    param: &mut LoginParam,
     was_autoconfig: bool,
 ) -> Result<()> {
+    // manually_set_param is used to check whether a particular setting was set manually by the user.
+    // If yes, we do not want to change it to avoid confusing error messages
+    // (you set port 443, but the app tells you it couldn't connect on port 993).
+    let manually_set_param = LoginParam::from_database(context, "").await;
+
     let mut smtp = Smtp::new();
     /* try to connect to SMTP - if we did not got an autoconfig, the first try was SSL-465 and we do a second try with STARTTLS-587 */
-    if try_smtp_one_param(context, &param, &mut smtp).await.is_ok() {
+    if try_smtp_one_param(context, param, &mut smtp).await.is_ok() {
         return Ok(());
     }
     if was_autoconfig {
         return Ok(());
     }
     progress!(context, 850);
-    param.server_flags &= !(DC_LP_SMTP_SOCKET_FLAGS as i32);
-    param.server_flags |= DC_LP_SMTP_SOCKET_STARTTLS as i32;
-    param.send_port = 587;
 
-    if try_smtp_one_param(context, &param, &mut smtp).await.is_ok() {
+    if manually_set_param.server_flags & (DC_LP_SMTP_SOCKET_FLAGS as i32) == 0 {
```

Comment:
Does someone know why DC_LP_SMTP_SOCKET_FLAGS is usize? It is casted to i32 everywhere it is used.

Reply:
reminiscent of the c2rust transform i think.